### PR TITLE
Adds outbox metadata fields

### PR DESCRIPTION
**Pull request by elena-kolevska**
# Description
Adds metadata fields needed for the outbox pattern

## Checklist

* [X] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: .build-tools/pkg/metadataschema/validators.go [hunk 1]
```diff
@@ -111,6 +111,32 @@ func (c *ComponentMetadata) AppendBuiltin() error {
...
+			c.Metadata = append(c.Metadata,
+				Metadata{
+					Name:        "outboxPublishPubsub",
+					Type:        "string",
+					Description: "Sets the name of the pub/sub component to deliver the notifications when publishing state changes",
+				},
+				Metadata{
+					Name:        "outboxPublishTopic",
+					Type:        "string",
+					Description: "Sets the topic that receives the state changes on the pub/sub configured with outboxPublishPubsub. The message body will be a state transaction item for an insert or update operation",
```
Comment (ItalyPaleAle) [inline .build-tools/pkg/metadataschema/validators.go:124]:
```suggestion
					Description: `For outbox. Sets the topic that receives the state changes on the pub/sub configured with "outboxPublishPubsub". The message body will be a state transaction item for an insert or update operation`,
```

File: .build-tools/pkg/metadataschema/validators.go [hunk 2]
```diff
@@ -111,6 +111,32 @@ func (c *ComponentMetadata) AppendBuiltin() error {
...
+				},
+				Metadata{
+					Name:        "outboxPubsub",
+					Type:        "string",
+					Description: "Sets the pub/sub component used by Dapr to coordinate the state and pub/sub transactions. If not set, the pub/sub component configured with outboxPublishPubsub is used. This is useful if you want to separate the pub/sub component used to send the notification state changes from the one used to coordinate the transaction",
```
Comment (ItalyPaleAle) [inline .build-tools/pkg/metadataschema/validators.go:129]:
```suggestion
					Description: `For outbox. Sets the pub/sub component used by Dapr to coordinate the state and pub/sub transactions. If not set, the pub/sub component configured with "outboxPublishPubsub" is used. This is useful if you want to separate the pub/sub component used to send the notification state changes from the one used to coordinate the transaction`,
```

File: .build-tools/pkg/metadataschema/validators.go [hunk 3]
```diff
@@ -111,6 +111,32 @@ func (c *ComponentMetadata) AppendBuiltin() error {
 				},
 			)
 		}
+		if slices.Contains(c.Capabilities, "transactional") {
+			c.Metadata = append(c.Metadata,
+				Metadata{
+					Name:        "outboxPublishPubsub",
+					Type:        "string",
+					Description: "Sets the name of the pub/sub component to deliver the notifications when publishing state changes",
```
Comment (ItalyPaleAle) [inline .build-tools/pkg/metadataschema/validators.go:119]:
```suggestion
					Description: "For outbox. Sets the name of the pub/sub component to deliver the notifications when publishing state changes",
```

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Why is this PR targeting release-1.12?

**Comment by elena-kolevska** [comment]
> Why is this PR targeting release-1.12?

Thanks for the review @ItalyPaleAle. I moved it to `master`.

**Review by ItalyPaleAle — APPROVED** [review]
LGTM thanks!

**Pull request merged**